### PR TITLE
Storm: Various Layout/SourcegraphWebApp cleanups

### PR DESCRIPTION
**Pull request by philipp-spiess**
What changed?

- Start using `useMatches` on the `Layout` component and added a handle to the Repo container to detect it properly
- Remove `GlobalContributions` from the new Layout (this is not necessary anymore in the post extensions controller world)
- Remove lazy loading of `SourcegraphWebApp` as it creates a very nasty waterfall that won't help us test Storms improvements. The bundle size should not grow by much anyway as it shares a lot of the dependencies with `LegacySourcegraphWebApp` that we already inline.

If I’m not mistaken with this change the user-experience of the `enableStorm` feature flag on or off should be identical. 😅   Maybe we can enable it on S2?

## Test plan

- CI is green
- The thing still starts so 🤷 

## App preview:

- [Web](https://sg-web-ps-layout-data-hoisting.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.28% (+7.83 kb) 🔺 | 0.02% (+2.99 kb) | -0.04% (-4.84 kb) | 0.26% (+2) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/layout-data-hoisting/compare-report.html;tab=live_object) for a full comparison between the commits 3d04cf3 and ad8dd3f or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/SourcegraphWebApp.tsx [hunk 1]
```diff
@@ -401,15 +405,29 @@ export const SourcegraphWebApp: React.FC<SourcegraphWebAppProps> = props => {
...
+                    .filter(isTruthy),
+
+                {
+                    path: '*',
+                    element: <RepoContainer {...context} />,
+                    // In RR6, the useMatches hook will only give you the location that is matched
+                    // by the path rule and not the path rule instead. Since we need to be able to
+                    // detect if we're inside the repo container reliably inside the Layout, we
+                    // expose this information in the handle object instead.
+                    handle: { isRepoContainer: true },
```
Comment (philipp-spiess) [inline client/web/src/SourcegraphWebApp.tsx:428]: cc @valerybugakov this is what I meant in today's call about how to detect the repo container vs. any other route

File: client/web/src/SourcegraphWebApp.tsx [hunk 2]
```diff
@@ -402,11 +403,14 @@ export const SourcegraphWebApp: React.FC<SourcegraphWebAppProps> = props => {
                 />
             ),
             children: props.routes
+                // Remove routes that are already migrated
+                .filter(route => route.path !== PageRoutes.RepoContainer)
```
Comment (valerybugakov) [inline client/web/src/SourcegraphWebApp.tsx:407]: Could you elaborate on why we need it? Currently running the app with the Storm flag enabled I get an error when I visit the repo page.

File: client/web/src/routes.tsx [hunk 3]
```diff
@@ -173,6 +174,11 @@ export const routes: readonly LayoutRouteProps[] = (
         {
             path: PageRoutes.RepoContainer,
             render: props => <RepoContainer {...props} />,
+            // In RR6, the useMatches hook will only give you the location that is matched
+            // by the path rule and not the path rule instead. Since we need to be able to
+            // detect if we're inside the repo container reliably inside the Layout, we
+            // expose this information in the handle object instead.
+            handle: { isRepoContainer: true },
```
Comment (valerybugakov) [inline client/web/src/routes.tsx:181]: Nice! I see. If there will be more cases where we will rely on the handle to figure out the current route, we can move the safety logic into a helper.

```tsx
const isXXXPage =
        routeMatches.some(
            routeMatch =>
                routeMatch.handle &&
                typeof routeMatch.handle === 'object' &&
                Object.hasOwn(routeMatch.handle, 'XXX')
        ) ?? false

File: client/web/src/Layout.tsx [hunk 4]
```diff
@@ -255,11 +262,6 @@ export const Layout: React.FC<LegacyLayoutProps> = props => {
                     />
                 </Panel>
             )}
-            <GlobalContributions
```
Comment (valerybugakov) [inline client/web/src/Layout.tsx:258]: Nice!

**Review by valerybugakov — APPROVED** [review]

File: client/web/src/SourcegraphWebApp.tsx (hunk 1 above)
Comment (valerybugakov) [reply client/web/src/SourcegraphWebApp.tsx]: Yeah, it's really nice!

File: client/web/src/SourcegraphWebApp.tsx (hunk 2 above)
Comment (philipp-spiess) [reply client/web/src/SourcegraphWebApp.tsx:407]: Oh, leftover that I forgot to delete, sorry! I duplicated repo container

**Pull request merged**